common/views/components/SearchForm/SearchForm.tsx

```diff
@@ -136,74 +124,38 @@ const SearchForm: FunctionComponent<Props> = ({
   }, [query]);
 
   useEffect(() => {
-    if (portalSortOrder !== routeProps.sortOrder) {
+    if (portalSortOrder !== sortOrder) {
       submit();
     }
   }, [portalSortOrder]);
 
   function updateUrl(form: HTMLFormElement) {
-    const languages = getFilterItemSelected(form, 'languages');
-    const workType = getFilterItemSelected(form, 'workType');
-    const subjectsLabel = getFilterItemSelected(form, 'subjects.label');
-    const genresLabel = getFilterItemSelected(form, 'genres.label');
-    const contributorsAgentLabel = getFilterItemSelected(
-      form,
-      'contributors.agent.label'
-    );
-    const sortOrder = portalSortOrder;
-    const sort =
-      sortOrder === 'asc' || sortOrder === 'desc' ? 'production.dates' : null;
-    const search = inputValue(form['search']);
-    const imagesColor = getSelectedFilterColor(form);
-
-    const itemsLocationsLocationType =
-      form['items.locations.locationType'] instanceof window.HTMLInputElement
-        ? form['items.locations.locationType'].checked
-          ? form['items.locations.locationType'].value.split(',')
-          : []
-        : [];
+    const formData = new FormData(form);
+    // see: https://github.com/microsoft/TypeScript/issues/30584
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const url = new URLSearchParams(formData);
 
-    // Location type
-    const itemsLocationsType = (
-      nodeListValueToArray(form['items.locations.type']) || []
-    )
-      .filter(checkbox => checkbox.checked)
-      .map(checkbox => checkbox.value);
+    const params: ParsedUrlQuery = Array.from(url.entries()).reduce(
```

Comment:
Can this live in a utilities file + have some comments explaining what it does?

Reply:
🙌

Reply:
I've done this, I'll get on some tests after this PR.